var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -57,31 +59,76 @@ class Qmcpack(CMakePackage):
     # variant('+mixed', default=True, when='+cuda', description="...")
 
     # conflicts
-    conflicts('+soa', when='+cuda')
-    conflicts('^openblas+ilp64')
-    conflicts('^intel-mkl+ilp64')
+    conflicts(
+        '+phdf5',
+        when='~mpi',
+        msg='Parallel collective I/O requires MPI-enabled QMCPACK.\n \
+        Please add \'~phdf5\' to the Spack install line for serial QMCPACK.'
+    )
+    conflicts(
+        '+qe',
+        when='~mpi',
+        msg='QMCPACK QE variant requires MPI due to limitation in QE build system.\n \
+        Please add \'~qe\' to the Spack install line for serial QMCPACK.'
+    )
+    conflicts(
+        '+soa',
+        when='+cuda',
+        msg='QMCPACK SOA variant does not exist for CUDA'
+    )
+    conflicts(
+        '^openblas+ilp64',
+        msg='QMCPACK does not support OpenBLAS 64-bit integer variant'
+    )
+    conflicts(
+        '^intel-mkl+ilp64',
+        msg='QMCPACK does not support MKL 64-bit integer variant'
+    )
+
+    # CUDA conflicts taken from here
+    # https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html
+    cuda_warning = 'CUDA is not supported by this version of your compiler'
```

Comment:
Is this something that could (or should) be moved to the `cuda` package? I assume this affects all packages, not just qmcpack?

Reply:
This is a good point. I discovered these conflicts by accident. These are universal conflicts and not specific to QMCPACK. I digged into the CUDA package and was surprised the conflicts were absent from the CUDA package. I can remove this from the QMCPACK package and make a separate PR on the CUDA package. It would be helpful to get some buy-in from whomever owns the CUDA package.